pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -147,6 +152,104 @@ type ServiceInstancePort struct {
 	Protocol protocol.Instance `json:"protocol,omitempty"`
 }
 
+func (lb *ListenerBuilder) buildInboundHBONEListeners() []*listener.Listener {
+	vhost := &route.VirtualHost{
+		Name:    "connect",
+		Domains: []string{"*"},
+	}
+	for _, cc := range lb.buildInboundChainConfigs() {
+		// TODO passthrough
+		p := cc.port.TargetPort
+		name := fmt.Sprintf("inbound-hbone|%d", p)
+		vhost.Routes = append(vhost.Routes, &route.Route{
+			Match: &route.RouteMatch{
+				PathSpecifier: &route.RouteMatch_ConnectMatcher_{ConnectMatcher: &route.RouteMatch_ConnectMatcher{}},
+				Headers: []*route.HeaderMatcher{
+					istiomatcher.HeaderMatcher(":authority", fmt.Sprintf("*:%d", p)),
+				},
+			},
+			Action: &route.Route_Route{Route: &route.RouteAction{
+				UpgradeConfigs: []*route.RouteAction_UpgradeConfig{{
+					UpgradeType:   "CONNECT",
+					ConnectConfig: &route.RouteAction_UpgradeConfig_ConnectConfig{},
+				}},
+
+				ClusterSpecifier: &route.RouteAction_Cluster{Cluster: name},
+			}},
+		})
+	}
+	l := &listener.Listener{
+		Name:    "inbound-hbone",
+		Address: util.BuildAddress("0.0.0.0", 15008),
```

Comment:
should this be podip instead of wildcard

Reply:
None of the other listeners are (15006/15001), so I am not sure why this one would be ?

Reply:
i think 15008 is bind to port, like some real inbound listener

Reply:
Yes those all use wildcard